mmv1/third_party/terraform/resources/resource_compute_router_interface.go.erb

```diff
@@ -328,6 +329,11 @@ func resourceComputeRouterInterfaceDelete(d *schema.ResourceData, meta interface
 			ifaceFound = true
 			continue
 		} else {
+			// If this is a redundant interface,
+			// remove its reference from other interfaces as well
+			if iface.RedundantInterface == ifaceName {
+				iface.RedundantInterface = "";
```

Comment:
This is the only part I'm concerned about here. Why remove the interface if it's already marked Computed?

Reply:
TL;DR
The API automatically sets the redundant_interface field under an interface that interface has been set as redundant of another interface. Anyway, the API is not able to automatically clean up the redundant interface field, if the interface referenced in redundant_interface is removed.

Longer explanation with example:
Assume you create two interfaces. You insert the first one, then the other one.
On the second interface you set redundant_interface, referencing the one formerly created.
The API automatically sets redundant interface also on the first interface created, referencing the second one, so each interface is set as the redundant of the other. This is why I put Optional and Computed equal to true.
We now remove either the first or the second interface. The API is not automatically able to look for redundant interface fields in other interfaces on that router that reference the interface we're removing. As such, we've to look for it and remove it manually.